k8s-operator/apis/v1alpha1/types_proxygroup.go

```diff
@@ -122,14 +127,24 @@ type TailnetDevice struct {
 }
 
 // +kubebuilder:validation:Type=string
-// +kubebuilder:validation:Enum=egress;ingress
+// +kubebuilder:validation:Enum=egress;ingress;kube-apiserver
 type ProxyGroupType string
 
 const (
-	ProxyGroupTypeEgress  ProxyGroupType = "egress"
-	ProxyGroupTypeIngress ProxyGroupType = "ingress"
+	ProxyGroupTypeEgress              ProxyGroupType = "egress"
+	ProxyGroupTypeIngress             ProxyGroupType = "ingress"
+	ProxyGroupTypeKubernetesAPIServer ProxyGroupType = "kube-apiserver"
 )
 
 // +kubebuilder:validation:Type=string
 // +kubebuilder:validation:Pattern=`^[a-z0-9][a-z0-9-]{0,61}$`
 type HostnamePrefix string
+
+// KubeAPIServerConfig contains configuration specific to the kube-apiserver ProxyGroup type.
+type KubeAPIServerConfig struct {
+	// AuthMode enables auth mode for the API Server proxy. In auth mode,
+	// requests from the tailnet proxied over to the Kubernetes API server
+	// are additionally impersonated using the sender's tailnet identity.
```

Comment:
Have we thought about the impersonation model with the Tailscale Services? I assume that it will _never_ be the case that grants can be tied to Tailscale Services (rather than nodes by node tag)?
I guess this might then feel a bit inconsistent- the proxy's name/endpoint will be the Service name, but the impersonation will be configured using the node tags.

Reply:
I guess that might mean that in practice this ProxyGroup type will always be somewhat a special case